tools/llvm-spirv/llvm-spirv.cpp

```diff
@@ -824,5 +830,39 @@ int main(int Ac, char **Av) {
                 << ", size in bytes = " << SpecConst.Size
                 << ", type = " << SpecConst.Type << "\n";
   }
+
+  if (SPIRVPrintReport) {
+    std::ifstream IFS(InputFile, std::ios::binary);
+    int ErrCode = 0;
+    std::optional<SPIRV::SPIRVModuleReport> BinReport =
+        SPIRV::getSpirvReport(IFS, ErrCode);
+    if (!BinReport) {
+      std::cout << "Invalid SPIR-V binary, error code is " << ErrCode << "\n";
+      return -1;
+    }
+
+    SPIRV::SPIRVModuleTextReport TextReport =
+        SPIRV::formatSpirvReport(BinReport.value());
+
+    std::cout << "SPIR-V module report:"
+              << "\n Version: " << TextReport.Version
+              << "\n Memory model: " << TextReport.MemoryModel
+              << "\n Addressing model: " << TextReport.AddrModel << "\n";
+
+    std::cout << " Number of capabilities: " << TextReport.Capabilities.size()
+              << "\n";
+    for (auto Capability : TextReport.Capabilities)
+      std::cout << "  Capability: " << Capability << "\n";
+
+    std::cout << " Number of extensions: " << TextReport.Extensions.size()
+              << "\n";
+    for (auto Extension : TextReport.Extensions)
+      std::cout << "  Extension: " << Extension << "\n";
+
+    std::cout << " Number of extension imports: "
+              << TextReport.ExtendedInstructionSets.size() << "\n";
+    for (auto ExtendedInstructionSet : TextReport.ExtendedInstructionSets)
+      std::cout << "  Extension Import: " << ExtendedInstructionSet << "\n";
```

Comment:
```suggestion
      std::cout << "  Extended Instruction Set: " << ExtendedInstructionSet << "\n";
```

Reply:
This hasn't been done yet as far as I can tell?

Reply:
indeed, thank you